packages/samples/headless-ssr/src/app/generic/hooks/search-parameters.ts

```diff
@@ -31,44 +28,36 @@ export function useSyncSearchParameters({
   ssrState,
   controller,
 }: UseSyncSearchParametersProps) {
-  const historyRouter = useHistoryRouter();
+  const {push, replace} = useRouter();
+  const searchParams = useSearchParams();
   const state = useSearchParameters({ssrState, controller});
 
   // Update the search interface.
   useEffect(
-    () =>
-      controller &&
-      historyRouter.url?.searchParams &&
-      controller.synchronize(
-        CoveoNextJsSearchParametersSerializer.fromClientSideUrlSearchParams(
-          historyRouter.url.searchParams
-        ).coveoSearchParameters
-      ),
+    () => {
+      if (!controller) {
+        return;
+      }
+      if (searchParams === null) {
+        return;
+      }
+      controller.synchronize(searchParams.toString());
+    },
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [historyRouter.url?.searchParams]
+    [searchParams]
   );
 
   // Update the URL.
-  const correctedUrl = useMemo(() => {
-    if (!historyRouter.url) {
-      return null;
-    }
-    const newURL = new URL(historyRouter.url);
-    CoveoNextJsSearchParametersSerializer.fromCoveoSearchParameters(
-      state.parameters
-    ).applyToUrlSearchParams(newURL.searchParams);
-    return newURL.href;
-  }, [historyRouter.url, state.parameters]);
   useEffect(() => {
-    if (!correctedUrl || correctedUrl === historyRouter.url?.href) {
+    if (state.fragment === searchParams.toString()) {
       return;
     }
     const isInitialState = controller === undefined;
     if (isInitialState) {
-      historyRouter.replace(correctedUrl);
+      replace('?' + state.fragment, {shallow: true} as NavigateOptions);
     } else {
-      historyRouter.push(correctedUrl);
+      push('?' + state.fragment, {shallow: true} as NavigateOptions);
```

Comment:
Discovered this hidden workaround. The `shallow` option allows Next.js to update the parameters without reloading the page, but it isn't mentioned in their docs and isn't accepted by the function's type definition.

Reply:
Yeah.... That smells like an internal then? Ain't sure it's a good idea chief.

Reply:
Yeah, the alternative is to make our own hooks that also use NextJS's hooks underneath and try to hack things into working. We wouldn't need to use the `shallow` option though.

Reply:
For now, I'll close this PR, open a PR to remove `defineUrlManager`, and revisit after we've done a deeper investigation into URL management.